core/auth_common.go

```diff
@@ -0,0 +1,77 @@
+package core
+
+/**
+ * Copyright 2019 IBM All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import (
+	"fmt"
+	"net/http"
+)
+
+const (
+	// Supported authentication types.
+	AUTHTYPE_BASIC  = "basic"
+	AUTHTYPE_NOAUTH = "noauth"
+	AUTHTYPE_IAM    = "iam"
+	AUTHTYPE_ICP4D  = "icp4d"
```

Comment:
CP4D

Reply:
So... we've officially found a new name for "ICP4D"?  I thought we were going to do that in our meeting today.   My plan here was to use the same name we do everywhere else until we decide on the new official name.